metrics-datadog/src/main/java/org/coursera/metrics/datadog/DatadogReporter.java

```diff
@@ -74,36 +74,27 @@ public void report(SortedMap<String, Gauge> gauges,
                      SortedMap<String, Meter> meters,
                      SortedMap<String, Timer> timers) {
     final long timestamp = clock.getTime() / 1000;
-
-    List<String> newTags = tags;
-    if (tagsCallback != null) {
-      List<String> dynamicTags = tagsCallback.getTags();
-      if (dynamicTags != null && ! dynamicTags.isEmpty()) {
-        newTags = TagUtils.mergeTags(tags, dynamicTags);
-      }
-    }
-
     try {
       request = transport.prepare();
 
       for (Map.Entry<String, Gauge> entry : gauges.entrySet()) {
-        reportGauge(prefix(entry.getKey()), entry.getValue(), timestamp, newTags);
```

Comment:
We could pass the callback here to be cleaner and facilitate testing.  I opted for the easy way out